service/src/test/java/bio/terra/workspace/app/StartupInitializerTest.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.workspace.app;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import bio.terra.common.db.DataSourceManager;
+import bio.terra.common.migrate.LiquibaseMigrator;
+import bio.terra.landingzone.job.LandingZoneJobService;
+import bio.terra.landingzone.library.configuration.LandingZoneDatabaseConfiguration;
+import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
+import bio.terra.workspace.app.configuration.external.WorkspaceDatabaseConfiguration;
+import bio.terra.workspace.common.annotations.Unit;
+import bio.terra.workspace.service.job.StairwayInitializerService;
+import bio.terra.workspace.service.workspace.WsmApplicationService;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.ApplicationContext;
+
+@Unit
+@ExtendWith(MockitoExtension.class)
+class StartupInitializerTest {
+
+  @Mock private ApplicationContext context;
+  @Mock private DataSourceManager dataSourceManager;
+  @Mock private WorkspaceDatabaseConfiguration workspaceDatabaseConfiguration;
+  @Mock private LiquibaseMigrator liquibaseMigrator;
+  @Mock private StairwayInitializerService stairwayInitializerService;
+  @Mock private WsmApplicationService wsmApplicationService;
+  @Mock private FeatureConfiguration featureConfiguration;
+  @Mock private LandingZoneDatabaseConfiguration landingZoneDatabaseConfiguration;
+  @Mock private LandingZoneJobService landingZoneJobService;
+  @Mock private DataSource wsmDataSource;
+  @Mock private DataSource lzsDataSource;
+
+  private enum DatabaseInitializationInstruction {
+    INITIALIZE,
+    UPGRADE,
+    CONTINUE
+  }
+
+  @ParameterizedTest
```

Comment:
Why is this a parameterized test?

Reply:
Because when `StartupInitializer` runs, it may initialize the database, upgrade the database, or take no extra action.  Making this a parameterized test allows us to verify all of those pathways without duplicating code.

Reply:
Oh never mind, it's going to iterate over the enum values.